### ISSUE TITLE
W4.2: Commit accelerate configs (single/FSDP/DeepSpeed)

### ISSUE THREAD

**Issue by freddysongg** [freddysongg/LLM-W#31]
## PRD

**Problem:** `accelerate` requires config YAMLs specifying distributed strategy. Without committed configs, the FSDP path is undocumented and un-runnable.

**Goal:** Three committed accelerate configs covering single-GPU, FSDP 2-GPU, and DeepSpeed ZeRO-2 2-GPU.

## Technical Spec

Location: `configs/accelerate/`.

### `single_gpu.yaml`

```yaml
compute_environment: LOCAL_MACHINE
distributed_type: 'NO'
num_processes: 1
mixed_precision: bf16
```

### `fsdp_2gpu.yaml`

```yaml
compute_environment: LOCAL_MACHINE
distributed_type: FSDP
num_processes: 2
mixed_precision: bf16
fsdp_config:
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
  fsdp_transformer_layer_cls_to_wrap: Qwen2DecoderLayer
  fsdp_sharding_strategy: FULL_SHARD
  fsdp_state_dict_type: SHARDED_STATE_DICT
  fsdp_offload_params: false
gradient_checkpointing: true  # R-04 mitigation — pinned default
```

### `deepspeed_zero2_2gpu.yaml`

```yaml
compute_environment: LOCAL_MACHINE
distributed_type: DEEPSPEED
num_processes: 2
mixed_precision: bf16
deepspeed_config:
  zero_stage: 2
  offload_optimizer_device: none
  offload_param_device: none
gradient_checkpointing: true
```

### Dry-run validation

All three configs validated by:
```
accelerate launch --config_file configs/accelerate/<file> train.py --dry-run
```

## Acceptance Criteria

- [ ] 3 YAML configs committed
- [ ] All three enable `gradient_checkpointing: true` (spec §R-04)
- [ ] `accelerate launch --dry-run` validates each without error
- [ ] FSDP config pins `Qwen2DecoderLayer` as wrap class
- [ ] Documented in a new `configs/accelerate/README.md` — which to use when

## Dependencies

- **Blocked by:** #30 (accelerate integrated)
- **Blocks:** #34 (cloud FSDP run uses fsdp_2gpu.yaml)

## References

- Spec: `docs/plans/2026-04-13-llm-w-improvement-plan-v4.md` §W4.2
- Spec risk: §R-04 (gradient checkpointing default)

**Comment by freddysongg** [comment]
Landed in commit f0c2d4a. Added:
- `configs/accelerate/single_gpu.yaml` — single-device CPU/CUDA/MPS
- `configs/accelerate/fsdp_2gpu.yaml` — FSDP 2-GPU with Qwen2DecoderLayer wrap policy
- `configs/accelerate/deepspeed_zero2_2gpu.yaml` — DeepSpeed ZeRO-2 2-GPU
- `configs/accelerate/README.md` — invocation docs

All three set `gradient_checkpointing: true` per R-04. Usage: `accelerate launch --config_file configs/accelerate/<mode>.yaml`.
